modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLViewDisplayContext.java

```diff
@@ -245,28 +245,18 @@ public String getSelectFileEntryTypeURL() throws WindowStateException {
 		).buildString();
 	}
 
-	public String getSelectFolderURL() throws WindowStateException {
+	public String getSelectFolderURL() throws PortalException {
 		ItemSelector itemSelector =
 			(ItemSelector)_httpServletRequest.getAttribute(
 				ItemSelector.class.getName());
 
-		FolderItemSelectorCriterion folderItemSelectorCriterion =
-			new FolderItemSelectorCriterion();
+		FolderItemSelectorURLProvider folderItemSelectorURLProvider =
+			new FolderItemSelectorURLProvider(
+				_httpServletRequest, itemSelector);
 
-		folderItemSelectorCriterion.setDesiredItemSelectorReturnTypes(
-			new FolderItemSelectorReturnType());
-		folderItemSelectorCriterion.setFolderId(getFolderId());
-		folderItemSelectorCriterion.setRepositoryId(
-			_dlAdminDisplayContext.getSelectedRepositoryId());
-		folderItemSelectorCriterion.setSelectedFolderId(getFolderId());
-		folderItemSelectorCriterion.setSelectedRepositoryId(
-			_dlAdminDisplayContext.getSelectedRepositoryId());
-
-		PortletURL portletURL = itemSelector.getItemSelectorURL(
-			RequestBackedPortletURLFactoryUtil.create(_renderRequest),
-			"itemSelected", folderItemSelectorCriterion);
-
-		return portletURL.toString();
+		return folderItemSelectorURLProvider.getSelectMoveToFolderURL(
+			_dlAdminDisplayContext.getSelectedRepositoryId(), getFolderId(),
+			DLFolderConstants.DEFAULT_PARENT_FOLDER_ID);
```

Comment:
When we use `DLFolderConstants.DEFAULT_PARENT_FOLDER_ID` as default value we are not allowing to move a file on the Home folder.